### ISSUE TITLE
Implement the domain-specific `DataPropositionInputs`

### ISSUE THREAD

**Issue by christoph-maurer** [building-envelope-data/database#53]
In order to facilitate cross-domain queries, the fields https://github.com/building-envelope-data/api/blob/589cafc3fa7653045d8658fd6a2f486bf9b51c7d/apis/database.graphql#L707-L759 would need to be implemented so that they become part of https://github.com/building-envelope-data/database/blob/7e82b9f8cebb9b2504d77038ef9dc66f1d47f5da/frontend/type-defs.graphqls#L641-L654 .

**Comment by simonwacker** [comment]
The use case is that a user wants to find components with specific optical, photovoltaic, calorimetric, hygrothermal, and geometric properties, for example, a certain g-value and a certain thickness. This is in relational database terms an inner join of different domains (optical, photovoltaic, ...) on component IDs.

I advice against just adding `gValues`, `thicknesses`, and so forth from other propositions to `DataPropositionInput` as from the syntax it doesn't suggest that the meaning is that kind of inner join. The reason is that the queries are called `all*Data` and not `allComponentsWithData`, so we ask for data in queries and not for components with certain data.

What would be a good design for inner joins? Maybe we can take inspiration from [Hasura](https://hasura.io/). They support inner joins:

- [Hasura Inner Join Guide](https://www.restack.io/docs/hasura-knowledge-hasura-inner-join-guide)
- [Hasura GraphQL Joins Explained](https://www.restack.io/docs/hasura-knowledge-hasura-graphql-joins)

Basically, we would need to query for components:

```[graphql]
query {
   components(where: data: { ...}) {
      ...
   }
}
```

Here, filtering by data from various domains makes sense in the sense of inner joins. The problem is that product data databases are designed to store data and not components, so the natural entrypoints are data.

**Comment by christoph-maurer** [comment]
@simon-wacker Thank you for the explanations and suggestions! We need to limit the effort for product databases. Therefore, we give this topic a low priority. In the meantime, software companies can query for data in each domain and make the join which components fulfill requirements in several domains.

**Comment by simonwacker** [comment]
Because there is no `allData` query anymore, this issue is "resolved".
